### PR TITLE
Fix className prop issue when using Tooltip

### PR DESCRIPTION
**Pull request by swillis12**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [X] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?
1. Cannot add className prop to Tooltip when using Typescript: https://github.com/ant-design/ant-design/issues/16194.
2. TS error is thrown when adding this className prop.

### 💡 Solution
Tooltip Props now extend React.HTMLAttributes<HTMLElement>.

### 📝 Changelog

- English Changelog:
- Chinese Changelog (optional):

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by swillis12** [comment]
Not sure if this is the correct fix, please advise if not.

**Comment by ztplz** [comment]
What confuse me is why you want to add `classname` to `tooltip`,  I am not sure if it will work.

**Comment by swillis12** [comment]
> What confuse me is why you want to add `classname` to `tooltip`, I am not sure if it will work.

It does work, it is for applying text-overflow style to the tooltip so I don't have to have an extra layer of DOM nesting.

File: components/tooltip/index.tsx [hunk 1]
```diff
@@ -81,8 +81,8 @@ const splitObject = (obj: any, keys: string[]) => {
   return { picked, omitted };
 };
 
-class Tooltip extends React.Component<TooltipProps, any> {
-  static defaultProps = {
+class Tooltip extends React.Component<TooltipProps & React.HTMLAttributes<HTMLElement>, any> {
+    static defaultProps = {
```
Comment (yesmeck) [inline components/tooltip/index.tsx]: Unnecessary indent.

**Review by yesmeck — CHANGES_REQUESTED** [review]

File: components/tooltip/index.tsx (hunk 1 above)
Comment (swillis12) [reply components/tooltip/index.tsx]: Thanks, I have fixed this in https://github.com/ant-design/ant-design/pull/16195/commits/fec88af3d342d719c2c4c0f7210702e7d1d015e6

**Comment by zombieJ** [comment]
merge master

**Pull request merged**